scheme/sevsnp/common.go

```diff
@@ -0,0 +1,199 @@
+func comidFromJson(buf []byte) (*comid.Comid, error) {
+	extractedCorim, err := corim.UnmarshalUnsignedCorimFromJSON(buf)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(extractedCorim.Tags) > 1 {
+		return nil, errors.New("too many tags")
+	}
+
+	extractedComid, err := corim.UnmarshalComidFromCBOR(
+		extractedCorim.Tags[0],
+		extractedCorim.Profile,
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return extractedComid, nil
+}
+
+func parseCertificateChainFromEvidence(tsm *tokens.TSMReport) (*sevsnp.CertificateChain, error) {
+	var certTable abi.CertTable
+
+	if err := certTable.Unmarshal(tsm.AuxBlob); err != nil {
+		return nil, err
+	}
+
+	return certTable.Proto(), nil
+}
+
+func readCert(cert []byte) ([]byte, error) {
+	if len(cert) == 0 {
+		return nil, errors.New("empty certificate")
+	}
+
+	block, _ := pem.Decode(cert)
+	if block == nil || block.Type != "CERTIFICATE" {
+		return nil, fmt.Errorf("failed to read certificate")
+	}
+	return block.Bytes, nil
+}
+
+func parseAttestationToken(token *proto.AttestationToken) (*tokens.TSMReport, error) {
+	var (
+		err error
+		tsm = new(tokens.TSMReport)
+	)
+
+	switch token.MediaType {
+	case EvidenceMediaTypeTSMCbor:
+		err = tsm.FromCBOR(token.Data)
+		if err != nil {
+			return nil, err
+		}
+	case EvidenceMediaTypeTSMJson:
+		err = tsm.FromJSON(token.Data)
+		if err != nil {
+			return nil, err
+		}
+	case EvidenceMediaTypeRATSd:
+		eat := make(map[string]interface{})
+
+		err = json.Unmarshal(token.Data, &eat)
+		if err != nil {
+			return nil, err
+		}
+
+		cmwBase64, ok := eat["cmw"].(string)
+		if !ok {
+			return nil, fmt.Errorf("cmw not found in eat")
+		}
+
+		cmwJson, err := base64.StdEncoding.DecodeString(cmwBase64)
+		if err != nil {
+			return nil, err
+		}
+
+		cmwCollection, err := cmw.NewCollection("tag:github.com,2025:veraison/ratsd/cmw")
+		if err != nil {
+			return nil, err
+		}
+		err = cmwCollection.UnmarshalJSON(cmwJson)
+		if err != nil {
+			return nil, err
+		}
+
+		cmwMonad, err := cmwCollection.GetCollectionItem("tsm-report")
+		if err != nil {
+			return nil, err
+		}
+
+		cmwType, err := cmwMonad.GetMonadType()
+		if err != nil {
+			return nil, err
+		}
+		if cmwType != "application/vnd.veraison.configfs-tsm+json" {
+			return nil, fmt.Errorf("unexpected CMW type: %s", cmwType)
+		}
+		cmwValue, err := cmwMonad.GetMonadValue()
+		if err != nil {
+			return nil, err
+		}
+
+		err = tsm.FromJSON(cmwValue)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		return nil, fmt.Errorf("unexpected media type: %s", token.MediaType)
+	}
+
+	return tsm, nil
+}
+
+// transformSVNtoTCB extracts TCB from the supplied SVN. SEV-SNP's TCB_VERSION
+// is a composite version; it's bitfield consisting of SVNs from various firmware components
+func transformSVNtoTCB(svn comid.SVN) (*kds.TCBParts, error) {
+	var (
+		tcbVersion uint64
+		err        error
+		tcbParts   kds.TCBParts
+	)
+
+	// ToDo: following is a circuitous way to obtain the 64-bit TCB integer value
```

Comment:
If this is a genuine TO DO, preferably create an issue and tag the issue here, please!